book/entities.rst

```diff
@@ -0,0 +1,1182 @@
+architecture. Luckily, you can use the features provided by the `OroMigrationBundle`_ to create
+separate migrations for each bundle.
+
+Organizing migrations is pretty easy if you follow some basic conventions:
+
+* Place all migrations under the ``Migrations/Schema/`` directory of your bundle.
+* In this directory, create one subdirectory per schema version.
+* Create as many migration classes as necessary inside a particular schema version directory (see
+  the example below).
+
+.. note::
+
+    The names of the schema version directories are compared to each other using PHP's
+    :phpfunction:`version_compare` function. So it's good practice to name them like ``v1_0``,
+    ``v2_0`` and so on.
+
+When a migration to a particular schema version is performed, all migration classes from the
+corresponding directory are evaluated and the contents of their ``up()`` method is executed. A
+class is treated as a migration class when it implements the
+:class:`Oro\\Bundle\\MigrationBundle\\Migration\\Migration` interface.
+
+For example, the migration class for the ``Hotel`` entity, has to look like this::
+
+    // src/Acme/DemoBundle/Migraions/Schema/v1_0/Hotel.php
+    namespace Acme\DemoBundle\Migraions\Schema\v1_0;
+
+    use Doctrine\DBAL\Schema\Schema;
+    use Oro\Bundle\MigrationBundle\Migration\Migration;
+    use Oro\Bundle\MigrationBundle\Migration\QueryBag;
+
+    class Hotel implements Migration
+    {
+        public function up(Schema $schema, QueryBag $queries)
+        {
+            $table = $schema->createTable('acme_hotel');
+            $table->addColumn('id', 'integer', ['autoincrement' => true]);
+            $table->addColumn('name', 'string', ['length' => 255]);
+            $table->setPrimaryKey(['id']);
+            $table->addIndex(['name'], 'hotel_name_idx', []);
+        }
+    }
+
+You can modify the database using the interface the Doctrine DBAL offers with its ``Schema`` class
+and you can also execute queries directly using the ``QueryBag`` if needed.
+
+Queries that are executed using the ``QueryBag`` are divided into two groups: use the
+:method:`Oro\\Bundle\\MigrationBundle\\Migration\\QueryBag::addPreQuery` method to add a query
+that is executed before schema changes from a migration class are performed. Queries scheduled with
+the :method:`Oro\\Bundle\\MigrationBundle\\Migration\\QueryBag::addPostQuery` method are executed
+after the schema has been modified.
+
+To actually load and apply the migrations to the existing database schema, you have to execute the
+``oro:migrations:load`` command:
+
+.. code-block:: bash
+
+    $ php app/console oro:migrations:load
+
+This command checks for present migration versions that are currently not reflected in the existing
+database schema and executes all missing migrations sequentially in ascending order.
+
+.. tip::
+
+    You can use the ``--dry-run`` option to see what would be executed and you can use the
```

Comment:
also `--show-queries` option can be used to see detailed information about each SQL query